closed/adds/jdk/src/windows/native/jdk/crypto/jniprovider/NativeCrypto_md.c

```diff
@@ -23,27 +23,53 @@
  */
 
 #include <windows.h>
+#include <stdio.h>
+#include <jni.h>
+#include <sys/stat.h>
 
 #include "NativeCrypto_md.h"
 
+#define JAVA_DLL "java.dll"
+
+/* Prototypes. */
+jboolean GetJREPath(char *path, jint pathsize);
+jboolean GetApplicationHome(char *buf, jint bufsize);
+int JLI_Snprintf(char* buffer, size_t size, const char* format, ...);
```

Comment:
These functions should be `static` to prevent clashes with the like-named versions elsewhere.

Reply:
Done